package/MDAnalysis/transformations/translate.py

```diff
@@ -60,8 +61,74 @@ def wrapped(ts):
             v = np.float32(vector)
             ts.positions += v
         else:
-            raise ValueError("{} vector in too short".format(vector))
+            raise ValueError("{} vector is too short".format(vector))
+        
+        return ts
+    
+    return wrapped
+
+def center_in_box(ag, center='geometry', point=None, pbc=None):
+    """
+    Translates the coordinates of a given :class:`~MDAnalysis.coordinates.base.Timestep`
+    instance so that the center of geometry/mass of the given :class:`~MDAnalysis.core.groups.AtomGroup`
+    is centered on the unit cell. The unit cell dimensions are taken from the input Timestep object or
+    can be defined using the `box` argument.
+    
+    Example
+    -------
+    
+    .. code-block:: python
+    
+        ag = u.residues[1].atoms
+        ts = MDAnalysis.transformations.center(ag,center='mass')(ts)
+    
+    Parameters
+    ----------
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    center: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    point: list, optional
+        overrides the unit cell center - the coordinates of the Timestep are translated so
+        that the center of mass/geometry of the given AtomGroup is aligned to this position
+        instead. Defined as a list of size 3. Example: [1, 2, 3]
+    pbc: bool or None, optional
+        If True, all the atoms from the given AtomGroup will be moved to the unit cell
+        before calculating the center of mass or geometry
+    
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.base.Timestep` object
+    """
+    
+    pbc_arg = pbc
+    try:
+        if center == 'geometry':
+            ag_center = ag.center_of_geometry(pbc=pbc_arg)
```

Comment:
This cannot be done in the parent function. The way it is written now means that the center is not updated every frame. What you can do, though, is to choose the method in the parent method:

```python
# In the parent function
if center =='geometry':
    center_method = functools.partial(ag.center_of_geometry, pbc=pbc_arg)
elif ...

# In the wrapped function
ag_center = center_method()
```